Demo/Common/IntegrationExamples/Components/ApplePayComponentAdvancedFlowExample.swift

```diff
@@ -0,0 +1,240 @@
+            message = "Cancelled"
+        } else {
+            message = error.localizedDescription
+        }
+        finalize(false, message)
+    }
+
+    private func finalize(_ success: Bool, _ message: String) {
+        applePayComponent?.finalizeIfNeeded(with: success) { [weak self] in
+            guard let self = self else { return }
+            self.dismissAndShowAlert(success, message)
+        }
+    }
+
+    internal func dismissAndShowAlert(_ success: Bool, _ message: String) {
+        presenter?.dismiss {
+            // Payment is processed. Add your code here.
+            let title = success ? "Success" : "Error"
+            self.presenter?.presentAlert(withTitle: title, message: message)
+        }
+    }
+
+    // MARK: - Presentation
+
+    private func present(_ component: PresentableComponent) {
+        guard component.requiresModalPresentation else {
+            presenter?.present(viewController: component.viewController, completion: nil)
+            return
+        }
+
+        let navigation = UINavigationController(rootViewController: component.viewController)
+        component.viewController.navigationItem.rightBarButtonItem = .init(barButtonSystemItem: .cancel,
+                                                                           target: self,
+                                                                           action: #selector(cancelPressed))
+        presenter?.present(viewController: navigation, completion: nil)
+    }
+
+    @objc private func cancelPressed() {
+        applePayComponent?.cancelIfNeeded()
+        presenter?.dismiss(completion: nil)
+    }
+
+}
+
+extension ApplePayComponentAdvancedFlowExample: PaymentComponentDelegate {
+
+    internal func didSubmit(_ data: PaymentComponentData, from component: PaymentComponent) {
+        let request = PaymentsRequest(data: data)
+        apiClient.perform(request) { [weak self] result in
+            self?.paymentResponseHandler(result: result)
+        }
+    }
+
+    internal func didFail(with error: Error, from component: PaymentComponent) {
+        finish(with: error)
+    }
+
+}
+
+extension ApplePayComponentAdvancedFlowExample: ActionComponentDelegate {
+
+    internal func didFail(with error: Error, from component: ActionComponent) {
+        finish(with: error)
+    }
+
+    internal func didComplete(from component: ActionComponent) {
+        finish(with: .received)
+    }
+
+    internal func didProvide(_ data: ActionComponentData, from component: ActionComponent) {
+        (component as? PresentableComponent)?.viewController.view.isUserInteractionEnabled = false
+        let request = PaymentDetailsRequest(
+            details: data.details,
+            paymentData: data.paymentData,
+            merchantAccount: ConfigurationConstants.current.merchantAccount
+        )
+        apiClient.perform(request) { [weak self] result in
+            self?.paymentResponseHandler(result: result)
+        }
+    }
+}
```

Comment:
No need for ActionComponentDelegate for Apple Pay:

```suggestion
```

Reply:
47a1fb69f153def70200c018673e639d6796d548